src/index.ts

```diff
@@ -119,6 +146,23 @@ function parseLineHeight(value: string) {
 	const parsed = parseFloat(value);
 	if (parsed.toString() === value) {
 		return parsed;
+	} else {
+		const match = /^(\+|-)?(\.)?/.exec(value) as RegExpMatchArray;
```

Comment:
We can probably reuse the above-mentioned regex that I asked you to cache here.

Also, the TypeScript compiler will infer the type, so you shouldn't need `as RegExpMatchArray` here.

Reply:
without the cast, the compiler will complain that the `match` could be `null`:
but the regular expression will always match, since it has 2 optional parts.

So either have this cast, or add a not-`null` check to convince the compiler that using `match[1]` is OK -- I decided to use the cast, but if you prefer the not-`null` check, I will change it.

Similar for those other casts above:
it's i.m.o. a matter of taste, if you use the cast or add an explicit check -- I prefer the cast for instances where I know that the explicit check is unnecessary, as it does not add any JavaScript code after the compilation

Reply:
Why are you so sure the value wouldn't be `null`?

Reply:
it checks for (a) a beginning "^" and (b) two optional groups, so as long as the string has a beginning it will match

Reply:
I'm not sure what's wrong w/ your compiler or at what point it's giving you these errors, but I can tell you that the TypeScript playground has no problems with the following code:
```ts
function foo(value: string) {
    const match = /^(foo)?(bar)?/.exec(value);
    return match[1];
}
```
If I mouse-over `const match` it tells me that `match` is a `RegExpExecArray`.
![image](https://user-images.githubusercontent.com/1058243/44050842-cf185a3c-9efd-11e8-82be-21bc7de296b3.png)
Are you seeing something different? Note that `RegExpExecArray` and `RegExpMatchArray` are two different interfaces.

Reply:
no you are right, it should be `RegExpExecArray` (since this was only for silencing the compiler I did not look closely enough)

still: for `exec()` my compiler shows me a return value of `RegExpExecArray | null`, so without the cast there would be an explicit check for non-`null`